datafusion/optimizer/src/eliminate_limit.rs

```diff
@@ -35,35 +39,98 @@ impl EliminateLimit {
     }
 }
 
+/// Ancestor indicates the current ancestor in the LogicalPlan tree
+/// when traversing down related to "eliminate limit".
+enum Ancestor {
+    /// Limit
+    FromLimit { skip: Option<usize> },
+    /// Other nodes that don't affect the adjustment of "Limit"
+    NotRelevant,
+}
+
+/// replaces LIMIT 0 with an [LogicalPlan::EmptyRelation]
+/// replaces LIMIT node whose ancestor LIMIT's skip is greater than or equal to current's fetch
+/// with an [LogicalPlan::EmptyRelation]
+fn eliminate_limit(
+    _optimizer: &EliminateLimit,
+    ancestor: &Ancestor,
+    plan: &LogicalPlan,
+    _optimizer_config: &OptimizerConfig,
+) -> Result<LogicalPlan> {
+    match plan {
```

Comment:
How about
```
match plan {
  LogicalPlan::Limit(Limit {skip, Some(fetch), input, ..}})...
}
```

Reply:
`SELECT * FROM (SELECT * FROM test_a LIMIT 5) AS test_b OFFSET 5;`
In the above query, the current plan is the one with only `fetch` (the subquery).

I thought when the current node is `Limit` with only `skip`, it should be handled the same as `Projection`, `Sort`.

Reply:
The ancestor LIMIT node is OFFSET 5 for query "SELECT * FROM (SELECT * FROM test_a LIMIT 5) AS test_b OFFSET 5". 
For this case, we should use 5 as ancestor's skip.

Reply:
> Limit with only skip should be handled here. For query: SELECT * FROM (SELECT * FROM test_a LIMIT 5) AS test_b OFFSET 5;
> 
> what do you think about this?

I see, I misunderstood your code.